backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -4,70 +4,160 @@
-    private int MAX_ROWS_ALLOWED_PER_CSV;
     private static final String CSV_API_VERSION = "2";
-    private static final Logger LOG = LoggerFactory.getLogger(RaceArrayConverter.class);
+    private static final Logger LOG = LoggerFactory.getLogger(DataHubUploaderService.class);
+    private static final Date FALLBACK_EARLIEST_DATE = Date.from(Instant.parse("2020-01-01T00:00:00Z"));
+
+    private final DataHubConfig _config;
+    private final TestEventRepository _testReportEventsRepo;
+    private final DataHubUploadRespository _dataHubUploadRepo;
 
-    private final TestEventRepository _repo;
     private String _fileContents;
     private String _nextTimestamp;
     private String _warnMessage;
-    private String _uploadResultId;
+    private String _resultJson;
     private int _rowCount;
 
-    public DataHubUploaderService(TestEventRepository _repo) {
-        this._repo = _repo;
-        this._nextTimestamp = "";
-        this._warnMessage = "";
-        this._uploadResultId = "";
-        this._rowCount = 0;
+    public DataHubUploaderService(DataHubConfig config,
+                                  TestEventRepository testReportEventsRepo,
+                                  DataHubUploadRespository dataHubUploadRepo) {
+        _config = config;
+        _testReportEventsRepo = testReportEventsRepo;
+        _dataHubUploadRepo = dataHubUploadRepo;
+        _nextTimestamp = "";
+        _warnMessage = "";
+        _resultJson = "";
+        _rowCount = 0;
+    }
+
+    // todo: move to these somewhere common
+    private static String dateToUTCString(Date d) {
+        final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+        return simpleDateFormat.format(d);
+    }
+
+    private static Date utcStringToDate(String s) {
+        return Date.from(Instant.parse(s));
     }
 
-    private void _createTestEventCSV(String lastEndCreateOn) throws IOException, DateTimeParseException, NoResultException {
+    private void sendSlackChannelMessage(String titleMsg, List<String> markupMsgs, Boolean separateMsgs) {
+        if (!_config.getSlackNotifyWebhookUrl().startsWith("https://hooks.slack.com/")) {
+            LOG.error("SlackChannelNotConfigured. Message not sent Title: '{}' Body: '{}", titleMsg, String.join("  ", markupMsgs));
+            return;
+        } else {
+            // log the result since slack may be down.
+            LOG.info("slackMessage Title: '{}'  Body: '{}'", titleMsg, String.join(" ", markupMsgs));
+        }
+        try {
+            RestTemplate restTemplate = new RestTemplate();
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
+
+            // json (escaping)
+            // see https://api.slack.com/messaging/webhooks
+            // see https://app.slack.com/block-kit-builder/
+            JSONArray blocks = new JSONArray();
```

Comment:
Would it make sense to create a class for this? Seems a little less error prone than writing the serializer manually.

Reply:
I think that's tabbed for follow-up in #457.